src/Raven.Server/Documents/Queries/MoreLikeThis/Corax/MoreLikeThis.cs

```diff
@@ -333,16 +199,33 @@ public override void SetMaxDocFreqPct(int maxPercentage)
     protected PriorityQueue<object[]> RetrieveTerms(long documentId)
     {
         var indexSearcher = _builderParameters.IndexSearcher;
-        IDictionary<string, Int> termFreqMap = new HashMap<string, Int>();
-        var indexEntry = indexSearcher.GetEntryReaderFor(documentId);
+        Dictionary<string, int> termFreqMap = new();
+        Page p = default;
+        var indexEntry = indexSearcher.GetEntryTermsReader(documentId, ref p);
 
-        for (var i = 0; i < _fieldNames.Length; i++)
+        var fields = _fieldNames.Select(x => indexSearcher.FieldCache.GetLookupRootPage(x)).ToHashSet();
+        
+        while (indexEntry.MoveNext())
         {
-            var fieldName = _fieldNames[i];
-            AddTermFrequencies(ref indexEntry, termFreqMap, fieldName);
-        }
+            if(fields.Contains(indexEntry.FieldRootPage) == false)
+                continue;
+            
+            InsertTerm(indexEntry.Current.Decoded(), indexEntry.Frequency);
 
+        }
+        
         return CreateQueue(termFreqMap);
+        
+        
+        void InsertTerm(ReadOnlySpan<byte> term, int freq)
+        {
+            var termAsString = Encoding.UTF8.GetString(term);
+            if (IsNoiseWord(termAsString)) // TODO optimize
```

Comment:
Maybe let's create a ticket for that with more defails?

Reply:
The idea is that we should skip creating the string instance. Very low priority